submissions/Ant-C-tech/friends-app/js/app.js

```diff
@@ -0,0 +1,288 @@
+                                <img class="card__img" src="${friend.picture.large}" alt="person photo">
+                                <span class="card-title card__title card__title-${friend.gender}">${friend.name.first} ${friend.name.last}</span>
+                            </div>
+                            <div class="card-content card__content">
+                                <p>Hi, I am <span class="card__contentData">${friend.dob.age}</span> years old and I live in <span class="card__contentData">${friend.location.city}, ${friend.location.country}</span>.</p>
+                                <p>Call me, I want to be your friend.</p>
+                                <a href="tel:${friend.phone}" class="card__contentData">${friend.phone}</a>
+                            </div>
+                                <div class="card-action">
+                                <a class="card__contentData card__contentData-mail href="mailto:${friend.email}">${friend.email}</a>
+                            </div>
+                        </div>`
+        FriendsScreen += card
+    })
+    FriendsScreen += '</div></div>'
+    return FriendsScreen
+}
+
+function showSearchBar() {
+    SEARCH.classList.add(SHOW_ELEM_PRIMARY_ANIMATION)
+    SEARCH.classList.add('showElem')
+    SEARCH_INPUT.addEventListener('input', toSearch)
+}
+
+//Function is waiting to get content for display and some classes of AnimateCSS to animation including animation speed (optional)
+function changeContent(content, show, hide, speed) {
+
+    //Preparation displayed content for animation of hiding
+    MAIN.classList.remove('scroll') //Hide scrollbar (for esthetic purpose)
+    if (speed) {
+        MAIN.classList.add(speed) // Adding class that will define speed of animation(optional)
+    }
+
+    //Hiding displayed content using AnimateCSS class
+    MAIN.classList.add(hide)
+
+    //Waiting for end of animation of hiding content...
+    MAIN.addEventListener('animationend', function() {
+            //...and then...
+            MAIN.innerHTML = '' //Clear content
+            MAIN.innerHTML = content //Add new content
+            MAIN.classList.remove(hide) //Clean up unnecessary class
+
+            //Start animation of showing new content
+            MAIN.classList.add(show)
+
+            //Waiting for end of animation of showing content...
+            MAIN.addEventListener('animationend', function() {
+                    //...and then...
+                    MAIN.classList.add('scroll') //Show scrollbar back
+                    MAIN.classList.remove(show) //Clean up unnecessary class
+                    if (speed) {
+                        MAIN.classList.remove(speed) //Clean up unnecessary class
+                    }
+
+                }, { once: true }) // Use listeners once and automatically removing after invoke
+        }, { once: true }) //in order not to accumulate listeners after each function's call
```

Comment:
I still don't understand what problem you trying to solve. I can assume, that you have weird side effects from a lot listeners. If so, I believe proper way will be add listener once so it will be re-used. And not add it every time.

So you will don't need `once` anymore.

Reply:
I have tried to use a bit my app with "once:true" and without "once:true". There are screenshots below.
I am not sure, but I can see that every call of function changeContent leave listeners and it seems to me it is not good. Maybe there is just different way to organise an animation, but I have no experience in that. It seems to me that I heard that JQuery do it in this way or something like that. I do not have many resourses to use, so I used this one [(https://www.youtube.com/watch?v=0TLnIQbk5qo&ab_channel=%D0%94%D0%BC%D0%B8%D1%82%D1%80%D0%B8%D0%B9%D0%9B%D0%B0%D0%B2%D1%80%D0%B8%D0%BA)]
 , not step by step, but as I could understand it.
Anyway, I left "once:true" one more time. It is not principal to me, I have just tried to explain my logic. If you say it is wrong I will remove them without any problem. Thank you for your patience.

![With once_true](https://user-images.githubusercontent.com/59659904/103174683-db6db500-486c-11eb-84e8-07a443df5a22.jpg)
![Without once_true___](https://user-images.githubusercontent.com/59659904/103174806-bf1e4800-486d-11eb-8852-585f5abe8023.jpg)

Reply:
Yes, you have multiple calls of multiple listeners.

But can you add listener once, during initialization of your application? You will need to rewrite your listener for this, ofc.

Reply:
Thank you for your advice.
I have rewrote my listener and got rid from one temporary listener, but I still need one temporary listen to clear unnecessary classes after animation. As alternative for listener I could use setTimeout but it works not so well like listener. 
Maybe using animation effects was not good idea especially if there is no posibility to use temporary listeners. What do you think about it because I really need help with this problem.